lib/cisco_node_utils/router_ospf_area.rb

```diff
@@ -0,0 +1,285 @@
+    # area 1.1.1.1 authentication message-digest
+    def authentication
+      auth = config_get('ospf_area', 'authentication', @get_args)
+      return default_authentication unless auth
+      auth.include?('message-digest') ? 'md5' : 'clear_text'
+    end
+
+    def authentication=(val)
+      state = val ? '' : 'no'
+      auth = (val == 'md5') ? 'message-digest' : ''
+      set_args_keys(state: state, auth: auth)
+      config_set('ospf_area', 'authentication', @set_args)
+    end
+
+    def default_authentication
+      config_get_default('ospf_area', 'authentication')
+    end
+
+    # CLI can be the following or none
+    # area 1.1.1.1 default-cost 1000
+    def default_cost
+      config_get('ospf_area', 'default_cost', @get_args)
+    end
+
+    def default_cost=(val)
+      state = val == default_default_cost ? 'no' : ''
+      cost = val == default_default_cost ? '' : val
+      set_args_keys(state: state, cost: cost)
+      config_set('ospf_area', 'default_cost', @set_args)
+    end
+
+    def default_default_cost
+      config_get_default('ospf_area', 'default_cost')
+    end
+
+    # CLI can be the following or none
+    # area 1.1.1.1 filter-list route-map aaa in
+    def filter_list_in
+      config_get('ospf_area', 'filter_list_in', @get_args)
+    end
+
+    def filter_list_in=(val)
+      return if filter_list_in == false && val == false
+      state = val ? '' : 'no'
+      rm = val ? val : filter_list_in
+      set_args_keys(state: state, route_map: rm)
+      config_set('ospf_area', 'filter_list_in', @set_args)
+    end
+
+    def default_filter_list_in
+      config_get_default('ospf_area', 'filter_list_in')
+    end
+
+    # CLI can be the following or none
+    # area 1.1.1.1 filter-list route-map bbb out
+    def filter_list_out
+      config_get('ospf_area', 'filter_list_out', @get_args)
+    end
+
+    def filter_list_out=(val)
+      return if filter_list_out == false && val == false
+      state = val ? '' : 'no'
+      rm = val ? val : filter_list_out
+      set_args_keys(state: state, route_map: rm)
+      config_set('ospf_area', 'filter_list_out', @set_args)
+    end
+
+    def default_filter_list_out
+      config_get_default('ospf_area', 'filter_list_out')
+    end
+
+    # range can take multiple values for the same vrf
+    # area 1.1.1.1 range 10.3.0.0/32
+    # area 1.1.1.1 range 10.3.0.1/32
+    # area 1.1.1.1 range 10.3.3.0/24
+    # area 1.1.1.1 range 10.3.0.0/16 not-advertise cost 23
+    # sometimes the not-advertise and cost are reversed in the
+    # show command for reasons unknown!
+    # area 1.1.1.1 range 10.3.0.0/16 cost 23 not-advertise
+    # use positional way of getting the values as it is
+    # simple and there are only two properties which are
+    # optional. ip is mandatory
+    # the return list is of the form [ip, not-advertise, cost]
+    # ex: [['10.3.0.0/16', true, '23'], ['10.3.0.0/32', true, false],
+    #      ['10.3.0.1/32', false, false], ['10.3.3.0/24', false, '450']]
```

Comment:
Actually this is a multiple array setter.
So the manifest is like:
[['10.3.0.0/16', true, '23'], ['10.3.0.0/32', true, false], ['10.3.0.1/32', false, false]]

The one you gave me above works only for one array but not for arrays within array. I agree the way I have done is not the best but it is simple. Also there is a show command issue where the it is shifting the properties for some unknown reason randomly.